src/util/event_loop_signal.hpp

```diff
@@ -21,11 +21,11 @@
 
 #include <realm/util/features.h>
 
-#if (defined(REALM_HAVE_UV) && REALM_HAVE_UV && !REALM_PLATFORM_APPLE) || REALM_PLATFORM_NODE
+#if (defined(REALM_HAVE_UV) && REALM_HAVE_UV && !REALM_PLATFORM_APPLE) || (defined(REALM_PLATFORM_NODE) && REALM_PLATFORM_NODE)
 #define REALM_USE_UV 1
 #endif
 
-#if REALM_USE_UV
+#if defined(REALM_USE_UV) && REALM_USE_UV
```

Comment:
IMO it'd be preferable to add an `#else` branch to the `#if` that defines `REALM_USE_UV` to set it to 0. Ditto for other similar defines (`REALM_USE_EPOLL` in external_commit_helper.hpp, and `REALM_USE_UV` in event_loop.cpp).